src/main/java/seedu/module/model/Model.java

```diff
@@ -52,36 +54,59 @@
     void setModuleBook(ReadOnlyModuleBook moduleBook);
 
     /**
-     * Returns the ModuleBook
+     * Returns the ModuleBook.
      */
     ReadOnlyModuleBook getModuleBook();
 
     /**
      * Returns true if a Module with the same identity as {@code Module} exists in the module book.
      */
-    boolean hasModule(Module module);
+    boolean hasModule(TrackedModule trackedModule);
 
     /**
      * Deletes the given module.
      * The module must exist in the module book.
      */
-    void deleteModule(Module target);
+    void deleteModule(TrackedModule target);
 
     /**
      * Adds the given module.
      * {@code module} must not already exist in the module book.
      */
-    void addModule(Module module);
+    void addModule(TrackedModule trackedModule);
 
     /**
-     * Returns an unmodifiable view of the filtered module list
+     * Returns an unmodifiable view of the filtered module list.
      */
-    ObservableList<Module> getFilteredModuleList();
+    ObservableList<TrackedModule> getFilteredModuleList();
+
+    /**
+     * Returns an unmodifiable view of the filtered module list within the archive list.
+     */
+    ObservableList<ArchivedModule> getFilteredArchivedModuleList();
+
+    /**
+     * Updates the filter of the filtered archived module list to filter by the given {@code predicate}.
+     *
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateFilteredArchivedModuleList(Predicate<Module> predicate);
 
     /**
      * Updates the filter of the filtered module list to filter by the given {@code predicate}.
      *
      * @throws NullPointerException if {@code predicate} is null.
      */
-    void updateFilteredModuleList(Predicate<Module> predicate);
+    void updateFilteredModuleList(Predicate<TrackedModule> predicate);
+
+    /**
+     * Returns an unmodifiable view of the current shown list.
+     */
+    ObservableList<Module> getDisplayedList();
+
+    /**
+     * Changes the current list to be shown.
+     */
+    void changeDisplayedList(String list);
```

Comment:
Can we use a Enum to determine which list to switch to, instead of using magic words "ArchivedList" and "TrackedList"